spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/post/SendResponseFilter.java

```diff
@@ -191,12 +191,24 @@ else if (context.getResponseGZipped() && isGzipRequested) {
 		}
 		finally {
 			/**
-			* Closing the wrapping InputStream itself has no effect on closing the underlying tcp connection since it's a wrapped stream. I guess for http
-			* keep-alive. When closing the wrapping stream it tries to reach the end of the current request, which is impossible for infinite http streams. So
-			* instead of closing the InputStream we close the HTTP response.
+			* We must ensure that the InputStream provided by our upstream pooling mechanism is ALWAYS closed
+		 	* even in the case of wrapped streams, which are supplied by pooled sources such as Apache's
+		 	* PoolingHttpClientConnectionManager. In that particular case, the underlying HTTP connection will
+		 	* web returned back to the connection pool iif either close() is explicitly called, a read
+			* error occurs, or the end of the underlying stream is reached. If, however a write error occurs, we will
+			* end up leaking a connection from the pool without an explicit close()
 			*
-			* @author Johannes Edmeier
```

Comment:
no need to remove the other authors name

Reply:
Sure - can blame him when my change goes awry ;)